allms/models/abstract.py

```diff
@@ -105,27 +106,49 @@ def generate(
         if output_data_model_class:
             return self._parse_model_output(model_responses)
         return model_responses
+    
+    def _extract_json_from_response(self, model_response_data: ResponseData) -> str:
+        search_results = self._json_pattern.findall(model_response_data.response)
+        
+        if len(search_results) == 0:
+            return model_response_data.response
+        
+        return search_results[0]
+
+    def _parse_response(
+            self, 
+            model_response_data: ResponseData
+        ) -> ResponseWithError:
+        raw_response = self._extract_json_from_response(model_response_data)
```

Comment:
Firstly, we should check if the `model_response_data` doesn't contain an error. In case when it contains an error we will overwrite it error about parsing which will be not an original error and users may have problems with catching errors like "Not allowed content" "Too long input" etc.

Reply:
This is not the case - see line 145. The output parsing logic is invoked only if there is no error returned from the LLM.